artemis-server/src/main/java/org/apache/activemq/artemis/core/server/cluster/qourum/QuorumVoteServerConnect.java

```diff
@@ -141,7 +141,11 @@ public synchronized Boolean getDecision() {
       return votesNeeded == 0;
    }
 
-   public void await(int latchTimeout, TimeUnit unit) throws InterruptedException {
+   public void await(int latchTimeout, TimeUnit unit, int maxClusterSize) throws InterruptedException {
```

Comment:
IMO this check should be moved into `QuorumManager`, given that:
- `maxClusterSize` is already owned and managed by it
- QuorumVoteServerConnect isn't aware of how many cluster members could perform the vote, just the expected number to unblock the quorum